service/src/main/java/com/commercetools/pspadapter/payone/mapping/MappingUtil.java

```diff
@@ -129,22 +129,30 @@ public static void mapCustomerToRequest(@Nonnull final AuthorizationRequest requ
 
     public static void mapShippingAddressToRequest(final AuthorizationRequest request, final Address shippingAddress) {
 
-        if(shippingAddress == null) {
-            throw new IllegalArgumentException("Missing shipping address details");
-        }
-
-        request.setShipping_firstname(shippingAddress.getFirstName());
-        request.setShipping_lastname(shippingAddress.getLastName());
-        request.setShipping_street(joinStringsIgnoringNull(Arrays.asList(shippingAddress.getStreetName(),
-            shippingAddress.getStreetNumber())));
-        request.setShipping_zip(shippingAddress.getPostalCode());
-        request.setShipping_city(shippingAddress.getCity());
-        request.setShipping_country(shippingAddress.getCountry().toLocale().getCountry());
-        request.setShipping_company(joinStringsIgnoringNull(Arrays.asList(shippingAddress.getCompany(),
-            shippingAddress.getDepartment())));
-
-        if (countriesWithStateAllowed.contains(shippingAddress.getCountry())) {
-            request.setShipping_state(shippingAddress.getState());
+        final String shippingCountry = shippingAddress.getCountry().toLocale().getCountry();
+        final String shipping_street = joinStringsIgnoringNull(asList(shippingAddress.getStreetName(),
+            shippingAddress.getStreetNumber()));
+
+        if(shippingAddress == null
+            || StringUtils.isBlank(shippingAddress.getPostalCode())
+            || StringUtils.isBlank(shippingCountry)
+            || StringUtils.isBlank(shipping_street)
+            || StringUtils.isBlank(shippingAddress.getCity())
```

Comment:
Are those 4 fields mandatory for PayPal and would result in an error from the payment provider side?

Reply:
Yes, These 4 fields are mandatory, if any field is missing in this 4 then it fails, tested and confirmed using prod credentials.